v1/services/storage/series_cursor.go

```diff
@@ -93,51 +95,27 @@ func newIndexSeriesCursorInfluxQLPred(ctx context.Context, predicate influxql.Ex
 		}
 	}
 
-	var mitr tsdb.MeasurementIterator
-	name, singleMeasurement := HasSingleMeasurementNoOR(p.measurementCond)
-	if singleMeasurement {
-		mitr = tsdb.NewMeasurementSliceIterator([][]byte{[]byte(name)})
-	}
-
 	sg := tsdb.Shards(shards)
-	p.sqry, err = sg.CreateSeriesCursor(ctx, tsdb.SeriesCursorRequest{Measurements: mitr}, opt.Condition)
-	if p.sqry != nil && err == nil {
-		// Optimisation to check if request is only interested in results for a
-		// single measurement. In this case we can efficiently produce all known
-		// field keys from the collection of shards without having to go via
-		// the query engine.
-		if singleMeasurement {
-			fkeys := sg.FieldKeysByMeasurement([]byte(name))
-			if len(fkeys) == 0 {
-				goto CLEANUP
-			}
-
-			fields := make([]field, 0, len(fkeys))
-			for _, key := range fkeys {
-				fields = append(fields, field{n: key, nb: []byte(key)})
-			}
-			p.fields = map[string][]field{name: fields}
-			return p, nil
-		}
-
-		var mfkeys map[string][]string
-		mfkeys, err = sg.FieldKeysByPredicate(opt.Condition)
-		if err != nil {
-			goto CLEANUP
-		}
-
+	if mfkeys, err := sg.FieldKeysByPredicate(opt.Condition); err == nil {
 		p.fields = make(map[string][]field, len(mfkeys))
+		fieldNames := []string{}
 		for name, fkeys := range mfkeys {
 			fields := make([]field, 0, len(fkeys))
 			for _, key := range fkeys {
 				fields = append(fields, field{n: key, nb: []byte(key)})
 			}
 			p.fields[name] = fields
+			fieldNames = append(fieldNames, name)
```

Comment:
`fieldNames` is a bad name here I think - should be something like `measurementNamesForFields`